src/box/xrow.c

```diff
@@ -1463,6 +1463,42 @@ xrow_decode_error(struct xrow_header *row)
 	box_error_set(__FILE__, __LINE__, code, error);
 }
 
+int
+xrow_decode_txn(const struct xrow_header *row, struct txn_request *request)
+{
+	assert(row->type == IPROTO_BEGIN);
+	memset(request, 0, sizeof(*request));
+
+	/** Request without extra options. */
+	if (row->bodycnt == 0)
+		return 0;
+
+	const char *begin = row->body[0].iov_base;
+	const char *end = begin + row->body[0].iov_len;
+	const char *pos = begin;
+	uint32_t map_size = mp_decode_map(&pos);
+	for (uint32_t i = 0; i < map_size; ++i) {
+		if (mp_typeof(*pos) != MP_UINT)
+			goto bad_msgpack;
+		uint64_t key = mp_decode_uint(&pos);
+		switch (key) {
+		case IPROTO_TXN_TIMEOUT:
+			if (mp_typeof(*pos) != MP_DOUBLE)
+				goto bad_msgpack;
```

Comment:
Please use iproto_key_type for checking argument type, like other similar functions do, - this will ease adding new keys.

Reply:
fixed